core/src/main/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilder.java

```diff
@@ -174,43 +180,62 @@ protected DefaultFlow(String name, MuleContext muleContext) {
-          return createReturnEventForParentFlowConstruct(result, event);
-        } catch (MessagingException e) {
-          e.setProcessedEvent(createReturnEventForParentFlowConstruct(e.getEvent(), event));
-          throw e;
-        } catch (Exception e) {
-          resetRequestContextEvent(event);
-          throw new DefaultMuleException(CoreMessages.createStaticMessage("Flow execution exception"), e);
-        }
+        return processBlockingSynchronous(event);
       } else {
         try {
-          return just(event).transform(this).block();
+          return just(event).transform(this).otherwise(EventDroppedException.class, ede -> empty()).block();
         } catch (Exception e) {
           throw rxExceptionToMuleException(e);
         }
       }
     }
 
+    /*
+     * Process flow using blocking {@link Processor#process(Event)} methods synchronously without apply any processing strategy.
+     * Note that individual processors internally may still you asynchronous non-blocking behaviour to achieved required
+     * functionality.
+     */
+    private Event processBlockingSynchronous(Event event) throws MessagingException, DefaultMuleException {
+      // TODO MULE-11023 Migrate transaction execution template mechanism to use non-blocking API
+      final Event newEvent = createMuleEventForCurrentFlow(event, event.getReplyToDestination(), event.getReplyToHandler());
+      try {
+        ExecutionTemplate<Event> executionTemplate =
+            createErrorHandlingExecutionTemplate(muleContext, this, getExceptionListener());
+        Event result = executionTemplate.execute(() -> pipeline.process(newEvent));
+        return createReturnEventForParentFlowConstruct(result, event);
+      } catch (MessagingException e) {
+        e.setProcessedEvent(createReturnEventForParentFlowConstruct(e.getEvent(), event));
+        throw e;
+      } catch (Exception e) {
+        resetRequestContextEvent(event);
+        throw new DefaultMuleException(CoreMessages.createStaticMessage("Flow execution exception"), e);
+      }
+    }
+
     @Override
     public Publisher<Event> apply(Publisher<Event> publisher) {
-      return from(publisher).flatMap(event -> {
-        Event request = createMuleEventForCurrentFlow(event, event.getReplyToDestination(), event.getReplyToHandler());
-        just(request).transform(processFlowFunction()).subscribe();
-        return Mono.from(request.getContext())
-            .map(r -> createReturnEventForParentFlowConstruct(r, event))
-            .otherwise(MessagingException.class, me -> {
-              me.setProcessedEvent(createReturnEventForParentFlowConstruct(me.getEvent(), event));
-              return error(me);
-            });
-      });
+      return from(publisher)
+          .doOnNext(assertStrarted())
+          .flatMap(event -> {
+            if (processingStrategyFactory instanceof LegacySynchronousProcessingStrategyFactory) {
```

Comment:
It would be more verbose to do it outside, and it has the same effect here given flatMap is required anyway.